### PR TITLE
Remove Fody dependency from Libplanet

### PR DESCRIPTION
**Pull request by limebell**
Remove `Fody.Equals` and classes using `[Equals]` attributed now implements `IEquatable<T>`.

Implementation `GetHashCode()` override method referenced:
https://github.com/Fody/Equals/blob/master/Equals.Fody/Injectors/GetHashCodeInjector.cs

due to compatibility.

### CODE REVIEW

**Comment by longfin** [comment]
We need to resolve conflicts 😢

**Review by longfin — APPROVED** [review]

File: Libplanet.Net/Libplanet.Net.csproj [hunk 1]
```diff
@@ -45,16 +44,10 @@
...
-    <PackageReference Include="Fody" Version="6.1.1">
-      <PrivateAssets>all</PrivateAssets>
-    </PackageReference>
     <PackageReference Include="Menees.Analyzers.2017" Version="2.0.3">
       <PrivateAssets>all</PrivateAssets>
     </PackageReference>
-    <PackageReference
-      Include="Microsoft.DotNet.Analyzers.Compatibility"
-      Version="0.2.12-alpha">
+    <PackageReference Include="Microsoft.DotNet.Analyzers.Compatibility" Version="0.2.12-alpha">
```
Comment (dahlia) [inline Libplanet.Net/Libplanet.Net.csproj:50]: It's trivial, but this change seems unnecessary.

File: Libplanet/Libplanet.csproj [hunk 2]
```diff
@@ -67,12 +66,9 @@ https://docs.libplanet.io/</Description>
     <PackageReference Include="BouncyCastle.NetCore" Version="1.8.6" />
     <PackageReference Include="Caching.dll" Version="1.4.0.1" />
     <PackageReference Include="Destructurama.Attributed" Version="2.0.0" />
-    <PackageReference Include="Equals.Fody" Version="4.0.1" />
-    <PackageReference Include="Fody" Version="6.1.1">
-      <PrivateAssets>all</PrivateAssets>
-    </PackageReference>
```
Comment (dahlia) [inline Libplanet/Libplanet.csproj:73]: These dependency removals should be recorded in the changelog.  Note that there is section *Dependencies*.

File: Libplanet/Tx/TxId.cs [hunk 3]
```diff
@@ -87,9 +86,21 @@ public ImmutableArray<byte> ByteArray
...
+
+        public bool Equals(TxId other)
+        {
+            return ByteArray.SequenceEqual(other.ByteArray);
+        }
+
+        public override bool Equals(object obj)
+        {
+            return obj is TxId other && Equals(other);
+        }
```
Comment (dahlia) [inline Libplanet/Tx/TxId.cs:101]: Trivial stylistic suggestions:

```suggestion
        public bool Equals(TxId other) =>
            ByteArray.SequenceEqual(other.ByteArray);

        public override bool Equals(object obj) =>
            obj is TxId other && Equals(other);
```

File: CHANGES.md [hunk 4]
```diff
@@ -18,12 +18,21 @@ To be released.
...
 
+ -  Inner logic of `ByteUtil.CalculateHashCode(byte[] bytes)` has modified.
+    [[#1866], [#1891]]
+
 ### Bug fixes
 
 ### Dependencies
 
+ -  No longer depend on *Fody*.  [[#1866], [#1891]]
+ -  No longer depend on *Fody.Equals*.  [[#1866], [#1891]]
```
Comment (dahlia) [inline CHANGES.md:29]: It's confusing, but the package is named the other way.  See also: <https://www.nuget.org/packages/Equals.Fody>.

```suggestion
 -  No longer depend on *Equals.Fody*.  [[#1866], [#1891]]
```

Comment (limebell) [reply CHANGES.md:29]: Ooops my bad 😥

Comment (dahlia) [reply CHANGES.md:29]: It's not your bad, but their bad!  😇

**Review by greymistcube — APPROVED** [review]

**Review by longfin — APPROVED** [review]

**Review by dahlia — APPROVED** [review]

**Pull request merged**